homeassistant/components/switch/__init__.py

```diff
@@ -85,6 +91,14 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a config entry."""
+    if entry.domain == DOMAIN:
+        registry = er.async_get(hass)
+        # Will raise vol.Invalid to abort config entry setup
```

Comment:
We should handle the error. vol.Invalid prints a stack trace.

I would have it raise `ConfigEntryNotReady` in case it's pointing at a device that has no unique ID and so is not in the state machine yet. That way it would try agian.

Reply:
Updated to return `False` to permanently fail if we fail to resolve the entity.

As discussed in Discord raising `ConfigEntryNotReady` for entities not in the state machine is not ideal.